src/mlpack/methods/ann/rbm/spike_slab_rbm_impl.hpp

```diff
@@ -105,9 +105,9 @@ RBM<InitializationRuleType, DataType, PolicyType>::Phase(
   DataType visiblePenaltyGrad = DataType(gradient.memptr() +
       weightGrad.n_elem + spikeBiasGrad.n_elem, 1, 1, false, false);
```

Comment:
can you try removing the variable `visiblePenaltyGrad` and set the value directly? Seems like visiblePenaltyGrad would just be a double.